bouncer/tests/all_swaps.ts

```diff
@@ -76,5 +76,38 @@ async function main() {
   appendSwap('SolUsdc', 'Eth', testVaultSwap);
   appendSwap('SolUsdc', 'Flip', testVaultSwap, true);
 
+  // Note: Test vault swap with invalid refund params to ensure the engine decoding fallback works.
+  // We observed this values in the wild and want to ensure that the engine can handle them gracefully.
+  allSwaps.push(
+    testVaultSwap(
+      'ArbEth',
+      'Eth',
+      undefined,
+      newCcmMetadata('Eth', undefined, undefined, '0xdeadbeef'),
```

Comment:
I've just realized that my suggestion on how to test this out doesn't work straight away. A parameter of the `newCcmMetadata` confused me (I will fix it). The problem is that the string passed here doesn't end up as the `cf_parameters` but rather as the `ccmAdditionalData`. Then the SDK encodes that into a valid `cf_parameters` in the contract call.
Getting around this is extremely annoying without manually implementing the call to the smart contract, but we might do that anyway to test the broker API.
I'll think about how to do it but I think these three extra swaps can be removed unfortunately. Sorry @Janislav for the confusion :sweat:

Reply:
Yeha pretty sure that's the only way. I added that note to this ticket: https://linear.app/chainflip/issue/PRO-1912/implement-bouncer-evm-vault-swaps-via-broker-api
I'll add those tests there, it will be possible after that ticket is implemented.